libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -1372,12 +1452,24 @@ - (void)identityCoordinator:(SFIdentityCoordinator *)coordinator didFailWithErro
 
         // No retry, as missing parameters are fatal
         [self log:SFLogLevelError format:@"Missing parameters attempting to retrieve identity data.  Error domain: %@, code: %ld, description: %@", [error domain], [error code], [error localizedDescription]];
-        [self revokeRefreshToken:[SFUserAccountManager sharedInstance].currentUser];
+        SFUserAccount *userAccount = [[SFUserAccountManager sharedInstance] accountForCredentials:coordinator.credentials];
+        [self revokeRefreshToken:userAccount];
         self.authError = error;
         [self execFailureBlocks];
     } else {
         [self showRetryAlertForAuthError:error alertTag:kIdentityAlertViewTag];
     }
 }
 
+- (SFOAuthCredentials *)createOAuthCredentials {
+    NSString *oauthClientId = [self oauthClientId];
+    NSString *identifier = [[SFUserAccountManager sharedInstance] uniqueUserAccountIdentifier:self.oauthClientId];
```

Comment:
Or I guess, based on the rest of the implementation, maybe we should just axe the local variable and directly use the property in the line below.